### PR TITLE
Run test n times

### PR DESCRIPTION
**Pull request by palashmarantas**
# Issue Link

Added support to save multiple test execution data

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: pkg/core/interfaces.go [hunk 1]
```diff
@@ -60,7 +60,7 @@ type YMLParserService interface {
 
 // TestStats is used for servicing stat collection
 type TestStats interface {
-	CaptureTestStats(pid int32) error
+	CaptureTestStats(pid int32, CollectTestStats bool) error
```
Comment (saurabh-prakash) [inline pkg/core/interfaces.go:63]: nit
```suggestion
	CaptureTestStats(pid int32, collectStats bool) error
```

File: pkg/service/teststats/teststats.go [hunk 2]
```diff
@@ -17,26 +17,26 @@ import (
...
 		},
-		ExecutionResultOutputChannel: make(chan core.ExecutionResult),
+		ExecutionResultOutputChannel: make(chan core.ExecutionResults),
 	}, nil
 
 }
 
 // CaptureTestStats combines the ps stats for each test
-func (s *ProcStats) CaptureTestStats(pid int32) error {
+func (s *ProcStats) CaptureTestStats(pid int32, CollectTestStats bool) error {
```
Comment (saurabh-prakash) [inline pkg/service/teststats/teststats.go:39]: nit: `collectStats` instead of `CollectTestStats`

File: pkg/service/teststats/teststats.go [hunk 3]
```diff
@@ -51,21 +51,24 @@ func (s *ProcStats) CaptureTestStats(pid int32) error {
 			s.logger.Errorf("no process stats found with pid %d", pid)
 		}
 		select {
-		case executionResult := <-s.ExecutionResultInputChannel:
+		case executionResults := <-s.ExecutionResultInputChannel:
 			// Refactor the impl of below 2 functions using generics when Go 1.18 arrives
 			// https://www.freecodecamp.org/news/generics-in-golang/
```
Comment (saurabh-prakash) [inline pkg/service/teststats/teststats.go:56]: These need to go inside the `if` block

File: pkg/service/teststats/teststats.go (hunk 2 above)
Comment (palashmarantas) [reply pkg/service/teststats/teststats.go]: done

File: pkg/service/teststats/teststats.go (hunk 3 above)
Comment (palashmarantas) [reply pkg/service/teststats/teststats.go]: done

File: pkg/core/interfaces.go (hunk 1 above)
Comment (palashmarantas) [reply pkg/core/interfaces.go]: done

File: pkg/core/lifecycle.go [hunk 4]
```diff
@@ -287,23 +287,32 @@ func (pl *Pipeline) Start(ctx context.Context) (err error) {
...
-			if testResult.Status == "failed" {
+		for _, result := range executionResults.Results {
+			for i := 0; i < len(result.TestPayload); i++ {
+				testResult := &result.TestPayload[i]
+				if testResult.Status == "failed" {
+					taskPayload.Status = Failed
+					break
+				}
+			}
+			if taskPayload.Status == "failed" {
```
Comment (rishabh-arya95) [inline pkg/core/lifecycle.go:315]: why is this required, you are marking it failed on L311

File: pkg/core/interfaces.go [hunk 5]
```diff
@@ -60,7 +60,7 @@ type YMLParserService interface {
 
 // TestStats is used for servicing stat collection
 type TestStats interface {
-	CaptureTestStats(pid int32) error
+	CaptureTestStats(pid int32, CollectStats bool) error
```
Comment (saurabh-prakash) [inline pkg/core/interfaces.go:63]:
```suggestion
	CaptureTestStats(pid int32, collectStats bool) error
```

File: pkg/service/teststats/teststats.go [hunk 6]
```diff
@@ -17,26 +17,26 @@ import (
...
 		},
-		ExecutionResultOutputChannel: make(chan core.ExecutionResult),
+		ExecutionResultOutputChannel: make(chan core.ExecutionResults),
 	}, nil
 
 }
 
 // CaptureTestStats combines the ps stats for each test
-func (s *ProcStats) CaptureTestStats(pid int32) error {
+func (s *ProcStats) CaptureTestStats(pid int32, CollectStats bool) error {
```
Comment (saurabh-prakash) [inline pkg/service/teststats/teststats.go:39]: lowercase variable name `collectStats`

File: pkg/core/lifecycle.go (hunk 4 above)
Comment (palashmarantas) [reply pkg/core/lifecycle.go]: outdated

File: pkg/core/interfaces.go (hunk 5 above)
Comment (palashmarantas) [reply pkg/core/interfaces.go:63]: done

File: pkg/service/teststats/teststats.go (hunk 6 above)
Comment (palashmarantas) [reply pkg/service/teststats/teststats.go:39]: done

Comment (saurabh-prakash) [reply pkg/service/teststats/teststats.go]: not done

**Review by saurabh-prakash — APPROVED** [review]

**Review by rishabh-arya95 — APPROVED** [review]

**Comment by saurabh-prakash** [comment]
changed base as per ci/cd changes
